src/sample_weight_audit/estimator_check.py

```diff
@@ -309,6 +312,20 @@ def multifit_over_weighted_and_repeated(
             est_weighted = clone(est).set_params(**extra_params_weighted)
             est_repeated = clone(est).set_params(**extra_params_repeated)
 
+        if shuffle:
+            rng = np.random.default_rng()
+            weighted_indices = rng.permutation(y_train.shape[0])
+            repeated_indices = rng.permutation(y_resampled_by_weights.shape[0])
+
+            X_train = np.take(X_train, weighted_indices, axis=0)
+            y_train = np.take(y_train, weighted_indices, axis=0)
```

Comment:
I find it curious that this can work without the following:

```suggestion
            y_train = np.take(y_train, weighted_indices, axis=0)
            sample_weight_train = np.take(sample_weight_train, weighted_indices)
```

Maybe it's just failing at detecting problems. The tests need to be updated to run with and without data shuffling.